include/circt/Dialect/Calyx/CalyxControl.td

```diff
@@ -34,6 +34,38 @@ def ControlOp : CalyxContainer<"control", [
   let verifier = "return ::verify$cppClass(*this);";
 }
 
+def IfOp : CalyxContainer<"if", [
+    ControlLike
+  ]> {
+  let summary = "Calyx If";
+  let arguments = (ins
+    I1:$cond,
+    FlatSymbolRefAttr:$groupName
+    );
+  let regions = (region SizedRegion<1>:$thenRegion, AnyRegion:$elseRegion);
+  let description = [{
+    The "calyx.if" operation represents and if-then-else construct for
+    conditionally executing two calyx groups. The operands to an if operation
+    is a boolean value and the group under which this value is evaluated.
```

Comment:
Nit: `... to an if operation is a 1-bit port and the group under which this port is driven.`

Reply:
About this... could the verifier check that the named group indeed assigns the port? Seems like we can put that check in without too much hassle and that might be a good behavior to lock in.

Reply:
While this makes good sense, I don't know whether this is an actual restriction of the current language spec - @cgyurgyik what does the native compiler do in this case?